tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -46,7 +48,14 @@ internal class SpanMessagePackFormatter : IMessagePackFormatter<TraceChunkModel>
         private readonly byte[] _runtimeIdNameBytes = StringEncoding.UTF8.GetBytes(Trace.Tags.RuntimeId);
         private readonly byte[] _runtimeIdValueBytes = StringEncoding.UTF8.GetBytes(Tracer.RuntimeId);
 
+        private readonly byte[] _environmentNameBytes = StringEncoding.UTF8.GetBytes(Trace.Tags.Env);
+        private readonly TransformedStringCache<byte[]> _environmentValueBytes = new(s => SerializeTagValue(s));
```

Comment:
Nitpicking - Maybe rename those to `_environmentValueBytesCache` as calling `_environmentValueBytes.GetTransformedValue` isn't obvious.

Reply:
Yeah, I meant to reuse `TransformedStringCache<T>` for another cache that returned strings instead of `byte[]` (think for example normalized or obfuscated strings), but ended up not doing that. I'll do some renaming :)

Reply:
Should be simpler now. Renamed many things, removed an extra layer of caching.